### ISSUE TITLE
Updating the "Contact Us" Form

### ISSUE THREAD

**Issue by Abi-Nada** [cds-snc/platform-forms-client#2476]
### Description:

Update the contact us form to improve information quality collected via it.

### Acceptance Criteria:
- Add the following fields:
    - [x] Department
    - [x] Branch
    - [x] Job title
- [x] Update the corresponding French form
~~- [ ] Update the link to the Support Form and the Contact form link within it in tandem~~
- [x] Update link on the 'Contact Us' button in marketing page
~~- [ ] Ensure all links are updated across the product~~

**Comment by connorscarolyns** [comment]
The link also needs to be updated on the get support form, and once that is updated, that form will have a new link and need to be updated on the contact us form... (it's a vicious cycle) :
![image.png](https://images.zenhubusercontent.com/620427f3846ebb620d78e582/d8043422-d23b-4ba3-a920-3c197ddab8ac)
![image.png](https://images.zenhubusercontent.com/620427f3846ebb620d78e582/505811f9-4bf4-4c21-b79f-b2c0dfde7b7d)

**Comment by jose-cds** [comment]
Updated version of the form. Sent to Anik for feedback and review

**Comment by jose-cds** [comment]
```
{
  "titleEn": "Contact us",
  "titleFr": "Contactez l'équipe de Formulaires GC",
  "introduction": {
    "descriptionEn": "Use this form to learn more about GC Forms and how it can work in your department.\n\nWe provide services to people who work in the government. We cannot give general advice to the public about the forms that we host. ",
    "descriptionFr": "Utilisez ce formulaire pour en savoir plus sur Formulaires GC et sur la façon dont votre département peut utiliser ce service.\n\nNous fournissons des services aux personnes qui travaillent au sein du gouvernement. Nous ne pouvons pas donner de conseils généraux au public sur les formulaires que nous hébergeons. Pour toute autre question ou pour tout problème technique, [veuillez contacter notre équipe de soutien.](https://forms-formulaires.alpha.canada.ca/fr/form-builder/support)"
  },
  "privacyPolicy": {
    "descriptionEn": "Learn about your [privacy and the Canadian Digital Service](https://digital.canada.ca/legal/privacy/).",
    "descriptionFr": "En savoir plus sur votre [vie privée et le Service numérique canadien](https://numerique.canada.ca/transparence/confidentialite/)."
  },
  "confirmation": {
    "descriptionEn": "Thank you for contacting the GC Forms team. We will reply to you via email within the next 3 business days.",
    "descriptionFr": "Merci d'avoir contacté l'équipe de Formulaires GC. Nous communiquerons avec vous dans les 3 jours ouvrables.",
    "referrerUrlEn": "",
    "referrerUrlFr": ""
  },
  "layout": [
    1,
    2,
    5,
    6,
    7,
    3,
    4
  ],
  "elements": [
    {
      "id": 1,
      "type": "textField",
      "properties": {
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Your name",
        "titleFr": "Votre nom",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": ""
      }
    },
    {
      "id": 2,
      "type": "textField",
      "properties": {
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Your email address",
        "titleFr": "Votre adresse courriel",
        "validation": {
          "type": "email",
          "required": true
        },
        "descriptionEn": "For example: your.name@canada.ca\n",
        "descriptionFr": "Par exemple: votre.nom@canada.ca"
      }
    },
    {
      "id": 5,
      "type": "textField",
      "properties": {
        "subElements": [],
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Your department",
        "titleFr": "Votre ministère ",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "placeholderEn": "",
        "placeholderFr": ""
      }
    },
    {
      "id": 6,
      "type": "textField",
      "properties": {
        "subElements": [],
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Your branch or sector",
        "titleFr": "Votre branche ou secteur",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "placeholderEn": "",
        "placeholderFr": ""
      }
    },
    {
      "id": 7,
      "type": "textField",
      "properties": {
        "subElements": [],
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Job title",
        "titleFr": "Titre du poste",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "placeholderEn": "",
        "placeholderFr": ""
      }
    },
    {
      "id": 3,
      "type": "checkbox",
      "properties": {
        "choices": [
          {
            "en": "Request a demo",
            "fr": "Demandez une démonstration du produit"
          },
          {
            "en": "Join our mailing list",
            "fr": "Inscrivez-vous à notre liste de diffusion"
          },
          {
            "en": "Give feedback",
            "fr": "Donnez votre avis"
          },
          {
            "en": "Other",
            "fr": "Autre"
          }
        ],
        "titleEn": "Reason for reaching out",
        "titleFr": "Raison pour laquelle vous nous contactez",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": ""
      }
    },
    {
      "id": 4,
      "type": "textArea",
      "properties": {
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Additional information (Provide more context)",
        "titleFr": "Informations supplémentaires (Veuillez fournir plus de contexte)",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": ""
      }
    }
  ]
}
```

**Comment by anikbrazeau** [comment]
{
  "titleEn": "Contact us",
  "titleFr": "Contactez-nous",
  "introduction": {
    "descriptionEn": "Use this form to learn more about GC Forms and how it can work in your department or agency.\n\nThe GC Forms team provides services to people who work in the government. We cannot give general advice to the public about the forms that we host.",
    "descriptionFr": "Utilisez ce formulaire pour en savoir plus sur Formulaires GC et sur la façon dont ce service peut être utile à votre ministère ou organisme.\n\nL'équipe Formulaires GC fournit des services aux personnes qui travaillent au sein du gouvernement. Nous ne pouvons pas donner de conseils généraux au public sur les formulaires que nous hébergeons."
  },
  "privacyPolicy": {
    "descriptionEn": "Learn about your [privacy and the Canadian Digital Service](https://digital.canada.ca/legal/privacy/).",
    "descriptionFr": "En savoir plus sur votre [vie privée et le Service numérique canadien](https://numerique.canada.ca/transparence/confidentialite/)."
  },
  "confirmation": {
    "descriptionEn": "## Thank you for contacting the GC Forms team\n\nWe will reply to you via email within the next 3 business days.",
    "descriptionFr": "## Merci d'avoir contacté l'équipe de Formulaires GC\n\nNous communiquerons avec vous dans les 3 jours ouvrables.",
    "referrerUrlEn": "",
    "referrerUrlFr": ""
  },
  "layout": [
    1,
    2,
    7,
    5,
    6,
    3,
    4
  ],
  "elements": [
    {
      "id": 1,
      "type": "textField",
      "properties": {
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Full name",
        "titleFr": "Nom complet",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "autoComplete": "name"
      }
    },
    {
      "id": 2,
      "type": "textField",
      "properties": {
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Email address",
        "titleFr": "Adresse courriel",
        "validation": {
          "type": "email",
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "autoComplete": "email"
      }
    },
    {
      "id": 5,
      "type": "textField",
      "properties": {
        "subElements": [],
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Department or agency",
        "titleFr": "Ministère ou organisme",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "placeholderEn": "",
        "placeholderFr": ""
      }
    },
    {
      "id": 6,
      "type": "textField",
      "properties": {
        "subElements": [],
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Branch or sector",
        "titleFr": "Direction ou secteur",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "placeholderEn": "",
        "placeholderFr": ""
      }
    },
    {
      "id": 7,
      "type": "textField",
      "properties": {
        "subElements": [],
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Job title",
        "titleFr": "Titre de poste",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": "",
        "placeholderEn": "",
        "placeholderFr": ""
      }
    },
    {
      "id": 3,
      "type": "checkbox",
      "properties": {
        "choices": [
          {
            "en": "Request a product demo",
            "fr": "Demandez une démonstration du produit"
          },
          {
            "en": "Join our mailing list",
            "fr": "Inscrivez-vous à notre liste de diffusion"
          },
          {
            "en": "Give feedback",
            "fr": "Donnez votre avis"
          },
          {
            "en": "Other",
            "fr": "Autre"
          }
        ],
        "titleEn": "Reason for reaching out",
        "titleFr": "Raison pour laquelle vous nous contactez",
        "validation": {
          "required": true
        },
        "descriptionEn": "",
        "descriptionFr": ""
      }
    },
    {
      "id": 4,
      "type": "textArea",
      "properties": {
        "choices": [
          {
            "en": "",
            "fr": ""
          }
        ],
        "titleEn": "Additional information",
        "titleFr": "Informations supplémentaires",
        "validation": {
          "required": false
        },
        "descriptionEn": "Provide more context",
        "descriptionFr": "Veuillez fournir plus de contexte"
      }
    }
  ]
}

**Comment by timarney** [comment]
@thiessenp-cds based on the above sample json do you have what you need to make updates on this?

**Comment by timarney** [comment]
Added

**Comment by timarney** [comment]
Hi All 

Here's the preview link for review

https://be7gyiry5fpbisf37p5l7drmx40myyyi.lambda-url.ca-central-1.on.aws/form-builder/support/contactus

**Comment by jose-cds** [comment]
Looks good for the most part, thank you! There is one small thing. I'm looking to update the language of the introduction. I'm getting the french reviewed. Once that's done, I'll share here.

**Comment by thiessenp-cds** [comment]
I merged the PR with the new Contact us fields. I can do the intro content change in a separate PR any time

**Comment by jose-cds** [comment]
Still waiting on the french text from the translation folks. Hopefully I'll hear back by later today. Will post here once done.

**Comment by jose-cds** [comment]
@thiessenp-cds got the french back! Here's the updated description in EN and FR.

EN

"Use this form to learn more about GC Forms and how it can work in your department or agency.

The GC Forms team provides services to people who work in the government. We cannot give general advice to the public about the forms that we host. For other enquiries, visit the [contacts page] (https://www.canada.ca/en/contact.html) to find the right department or agency to fulfill your request."


FR

"Utilisez ce formulaire pour en apprendre davantage sur Formulaires GC et sur la façon dont cet outil peut fonctionner au sein de votre ministère ou organisme.

L’équipe de Formulaires GC offre ses services au personnel du gouvernement. Nous ne pouvons pas conseiller le public au sujet des formulaires que nous hébergeons. Pour toute autre demande, rendez-vous sur la [page de contact](https://www.canada.ca/fr/contact.html) pour trouver le ministère ou organisme le plus à même de vous répondre."

Can you also set the email for the form to send to assistance+forms-formulaires@cds-snc.ca instead of mine?

**Comment by thiessenp-cds** [comment]
You can check out the changes at: https://mrmu7lwqsdmrghptvap6fyvfw40nzmoj.lambda-url.ca-central-1.on.aws/form-builder/support/contactus

**Comment by jose-cds** [comment]
Looks good to go to me!

**Comment by Abi-Nada** [comment]
Hi @thiessenp-cds !!

For "Update the link to the Support Form and the Contact form link within it in tandem": I think the initial assumption (which is wrong!) was that these were forms that we would need to delete and republish using the form builder....but I forgot that they were hardcoded in. **So we just need to update the "Contact us form" link on the Support page**

So sorry for the confusion!

**Comment by thiessenp-cds** [comment]
> Hi @thiessenp-cds !!
> 
> For "Update the link to the Support Form and the Contact form link within it in tandem": I think the initial assumption (which is wrong!) was that these were forms that we would need to delete and republish using the form builder....but I forgot that they were hardcoded in. **So we just need to update the "Contact us form" link on the Support page**
> 
> So sorry for the confusion!

Thanks! For the "contact us form" link on the Support page (screenshot just encase there is any confusion over which contact us link), would you know what the new link should be? Currently it points to our contact us page `/form-builder/support/contactus`.
![Screenshot 2023-08-16 at 10 11 22 AM](https://github.com/cds-snc/platform-forms-client/assets/107579368/25944e53-0f1c-4de9-8739-66cf6e975df0)

**Comment by jose-cds** [comment]
It should be pointing to the form you have been building Pete! I believe it is the same link as you have listed.

**Comment by timarney** [comment]
Shipped in 3.3.0

**Comment by Abi-Nada** [comment]
Ah sorry for the confusion - the earlier comments in this card made me think there would be a new link to the form you were building but that's not the case! I see the changes reflected correctly in prod, so we're good to close :)
